src/sage/combinat/designs/group_divisible_designs.py

```diff
@@ -90,36 +91,31 @@ def group_divisible_design(v,K,G,existence=False,check=False):
     blocks = None
 
     # from a (v+1,k,1)-BIBD
-    if (len(G) == 1 and
-        len(K) == 1 and
-        G[0]+1 in K):
+    if len(G) == 1 == len(K) and G[0] + 1 in K:
         from .bibd import balanced_incomplete_block_design
         k = K[0]
         if existence:
-            return balanced_incomplete_block_design(v+1,k,existence=True)
-        BIBD = balanced_incomplete_block_design(v+1,k)
+            return balanced_incomplete_block_design(v + 1, k, existence=True)
```

Comment:
could be simplified to `return balanced_incomplete_block_design(v + 1, k, existence= existence)`, no ?

Reply:
hmm, no. There no other return until many lines after.

Reply:
You are right.